tests/ctags/bug507864.c.tags

```diff
@@ -1,4 +1,5 @@
 # format=tagmanager
-ENTSEQNO�16�(seq)�0�FUNCSTS
-MEMTXT�16�(form_msg)�0�
-MEMTXT�1024�(form_msg)�0�
+ENTSEQNO�16�(eq)�0�FUNCSTS
+MEMTXT�16�(mail)�0�
+MEMTXT�1024�(orm_msg)�0�FUNCSTS
```

Comment:
I fixed the missing start letter (the collection didn't work correctly for ungetc()).

I'm not completely sure why the output is different than before - apparently the parser is terribly confused (both before and now too) and the output is strange.

Didn't mention it explicitly before but because of the sync with ctags, re-parsing now happens correctly in Geany (before wrong indexes were used so only one method was used for parsing). This is one change that might change the output but it could be something different too (haven't really checked as there's a difference just in this case which is wrong anyway so it doesn't matter much).